cmd/store/export.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright © 2023 OpenFGA
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/openfga/go-sdk/client"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+
+	"github.com/openfga/cli/internal/authorizationmodel"
+	"github.com/openfga/cli/internal/cmdutils"
+	"github.com/openfga/cli/internal/fga"
+	"github.com/openfga/cli/internal/output"
+	"github.com/openfga/cli/internal/storetest"
+	"github.com/openfga/cli/internal/tuple"
+)
+
+// buildStoreData compiles all the data necessary to output to file or stdout,
+// or returns an error if this was not successful.
+func buildStoreData(config fga.ClientConfig, fgaClient client.SdkClient) (*storetest.StoreData, error) {
+	// get the store
+	store, _ := fgaClient.GetStore(context.Background()).Execute()
+
+	var storeName string
+	if store != nil {
+		storeName = store.Name
+	}
+
+	// get the model
+	model, err := authorizationmodel.ReadFromStore(config, fgaClient)
+	if err != nil {
+		return nil, err //nolint:wrapcheck
+	}
+
+	authModel := authorizationmodel.AuthzModel{}
+	authModel.Set(*model.AuthorizationModel)
+
+	dsl, err := authModel.DisplayAsDSL([]string{"model"})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get model dsl: %w", err)
+	}
+
+	// get the tuples
+	rawTuples, err := tuple.Read(fgaClient, &client.ClientReadRequest{}, 10) //nolint:gomnd
```

Comment:
I've ignored it here, but wanted to highlight this magic number for "pages of tuples", is this number appropriate? Do we want to be able to override it with a flag, e.g. `--max-tuples` or similar?

Reply:
This has now been changed to accept a `--max-tuples` flag.